tasks/init.js

```diff
@@ -35,83 +35,202 @@ var config = defaults.getInstance('init')
     editorconfig: true
   });
 
-var check = yargs.createCheck()
-  // don't check if we are just accessing help
-  .withGate(function (argv) {
-    return !argv.help;
-  })
-  .withTest({
-    name: function(value) {
-      if (typeof value !== 'string') {
-        return 'name must be a string';
-      } else if (value.length === 0) {
-        return 'name must not be zero length';
-      }
-    },
-    version: function(value) {
-      if (typeof value !== 'string') {
-        return 'version must be a valid string';
-      } else if (!/\d+\.\d+\.\d+[-\w\d]*/.test(value)) {
-        return 'version must be 3-term semver string';
+var initOptionDefinitions = [
+  {
+    key: 'help',
+    value: {
+      describe: 'This help message',
+      alias   : [ 'h', '?' ],
+      boolean : true
+    }
```

Comment:
We need to consider the case of interactive menus.

Could we augment these definitions with a question form of the description?